<a href="https://colab.research.google.com/github/soniNiharika/EDA-project/blob/main/Copy_of_Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

Lets take a look at the data, which consist of two file

*   Play store data.csv
*   user_review



1. lets import the libraries required.

In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
% matplotlib inline

2. create a dataframe ('df_Play_Store') for storing data of Play store.csv file
and
a dataframe (df_User_Reviews) for storing data of 
User_Reviews.csv file

In [ ]:
# mount the drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# Define the path where the data is kept

path="/content/drive/MyDrive/Copy of Play Store Data.csv"

In [ ]:
# creating a dataframe for play store data.csv
df_Play_Store = pd.read_csv(path)

In [ ]:
# Define the path where the data us user review.csv kept

path="/content/drive/MyDrive/Copy of User Reviews.csv"

In [ ]:
# creating a dataframe for user reviews.csv

df_User_Reviews = pd.read_csv(path)

In [ ]:
# lets read the dataset df_Play_Store

df_Play_Store.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


## Play store data.csv: contains all the details of the applications on Google Play. There are 13 features that describe a given app.

In [ ]:
# lets read the dataset df_Play_Store

df_User_Reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     64295 non-null  object 
 1   Translated_Review       37427 non-null  object 
 2   Sentiment               37432 non-null  object 
 3   Sentiment_Polarity      37432 non-null  float64
 4   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), object(3)
memory usage: 2.5+ MB


## * User_reviews.csv: contains 64295 entries. There are three new features: Sentiment (Positive, Negative or Neutral), Sentiment Polarity and Sentiment Subjectivity.





# **2. Data cleaning**



In [ ]:
# check the columns that need cleaning
df_Play_Store.head(24)
                  

App  ...   Android Ver
0      Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                 Coloring book moana  ...  4.0.3 and up
2   U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                               Sketch - Draw & Paint  ...    4.2 and up
4               Pixel Draw - Number Art Coloring Book  ...    4.4 and up
5                          Paper flowers instructions  ...    2.3 and up
6             Smoke Effect Photo Maker - Smoke Editor  ...  4.0.3 and up
7                                    Infinite Painter  ...    4.2 and up
8                                Garden Coloring Book  ...    3.0 and up
9                       Kids Paint Free - Drawing Fun  ...  4.0.3 and up
10                            Text on Photo - Fonteee  ...    4.1 and up
11            Name Art Photo Editor - Focus n Filters  ...    4.0 and up
12                     Tattoo Name On My Photo Editor  ...    4.1 and up
13                              Mandala Coloring Book  ...    4.4 and up
14    3D Color Pixel by Number - Sandbox Art Coloring  ...    2.3 and up
15                    Learn To Draw Kawaii Characters  ...    4.2 and up
16       Photo Designer - Write your name with shapes  ...    4.1 and up
17                           350 Diy Room Decor Ideas  ...    2.3 and up
18                      FlipaClip - Cartoon animation  ...  4.0.3 and up
19                                       ibis Paint X  ...    4.1 and up
20                        Logo Maker - Small Business  ...    4.1 and up
21          Boys Photo Editor - Six Pack & Men's Suit  ...  4.0.3 and up
22            Superheroes Wallpapers | 4K Backgrounds  ...  4.0.3 and up
23                             Mcqueen Coloring pages  ...    4.1 and up

[24 rows x 13 columns]

In [ ]:
df_new=df_Play_Store[~df_Play_Store.duplicated()]

In [ ]:
df_new.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10358 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10358 non-null  object 
 1   Category        10358 non-null  object 
 2   Rating          8893 non-null   float64
 3   Reviews         10358 non-null  object 
 4   Size            10358 non-null  object 
 5   Installs        10358 non-null  object 
 6   Type            10357 non-null  object 
 7   Price           10358 non-null  object 
 8   Content Rating  10357 non-null  object 
 9   Genres          10358 non-null  object 
 10  Last Updated    10358 non-null  object 
 11  Current Ver     10350 non-null  object 
 12  Android Ver     10355 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [ ]:
df_new['Installs']

0            10,000+
1           500,000+
2         5,000,000+
3        50,000,000+
4           100,000+
            ...     
10836         5,000+
10837           100+
10838         1,000+
10839         1,000+
10840    10,000,000+
Name: Installs, Length: 10358, dtype: object

The 'Installs' column in 'play Store Data' is numerical type but it also contains [, and +] sign
because of which it is of string type.

So, firstly we have to convert this into numerical type.

In [ ]:
df_new['Size']

0                       19M
1                       14M
2                      8.7M
3                       25M
4                      2.8M
                ...        
10836                   53M
10837                  3.6M
10838                  9.5M
10839    Varies with device
10840                   19M
Name: Size, Length: 10358, dtype: object

The "Size" column contains an extra letter 'm' after the numerical value, which need to be corrected.


In [ ]:
type(df_new['Price'])
df_new['Price'].tail(50)

10791        0
10792        0
10793        0
10794        0
10795        0
10796        0
10797        0
10798    $1.04
10799        0
10800        0
10801        0
10802        0
10803        0
10804        0
10805        0
10806        0
10807        0
10808        0
10809        0
10810        0
10811        0
10812        0
10813        0
10814        0
10815        0
10816        0
10817        0
10818        0
10819        0
10820        0
10821        0
10822        0
10823        0
10824        0
10825        0
10826        0
10827        0
10828        0
10829        0
10830        0
10831        0
10832        0
10833        0
10834        0
10835        0
10836        0
10837        0
10838        0
10839        0
10840        0
Name: Price, dtype: object

The "Price" column also have a "$" sign which converts it into a string datatype.


These three features that we will be working with most frequently Installs, Size, and Price. A careful glance of the dataset reveals that some of these columns mandate data cleaning in order to be consumed by code we'll write later. Specifically, the presence of special characters (, $ +) and letters (M) in the Installs, Size, and Price columns make their conversion to a numerical data type difficult. Let's clean by removing these and converting each column to a numeric type.

In [ ]:
# make a list of characters that needed to be removed
char_to_remove = [',','$','+','M',]
# make a listof columns that needed to be cleaned
columns_to_clean = ['Price','Size','Installs']

# Loop for each column
for col in columns_to_clean:
    # Replace the characters in each column col with an empty string
    for char in char_to_remove:
        df_new[col] = df_new[col].str.replace(char, '')
   


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  # Remove the CWD from sys.path while we load stuff.


In [ ]:
df_new.sort_index()

App  ...         Android Ver
0         Photo Editor & Candy Camera & Grid & ScrapBook  ...        4.0.3 and up
1                                    Coloring book moana  ...        4.0.3 and up
2      U Launcher Lite – FREE Live Cool Themes, Hide ...  ...        4.0.3 and up
3                                  Sketch - Draw & Paint  ...          4.2 and up
4                  Pixel Draw - Number Art Coloring Book  ...          4.4 and up
...                                                  ...  ...                 ...
10836                                   Sya9a Maroc - FR  ...          4.1 and up
10837                   Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                             Parkinson Exercices FR  ...          2.2 and up
10839                      The SCP Foundation DB fr nn5n  ...  Varies with device
10840      iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[10358 rows x 13 columns]